### PR TITLE
Compliance

### PR DESCRIPTION
**Pull request by MikeOpenHWGroup**
This pull request brings in the latest RISC-V Compliance Test-suite and adds a specific-purpose MetricsCI regression for it.

### CODE REVIEW

**Comment by eroom1966** [comment]
Hi Mike,
I think you can remove any references to my own github for https://github.com/eroom1966/riscv-compliance
The changed are now merged into the root repository at
https://github.com/riscv/riscv-compliance

Other than that - it looks fine to me.

Thx
Lee

**Review by eroom1966 — CHANGES_REQUESTED** [review]
Hi Mike

All my changes are now on the main fork, please change any reference of
https://github.com/eroom1966/riscv-compliance
to
https://github.com/riscv/riscv-compliance

then we are good to go

Thx
Lee

**Comment by MikeOpenHWGroup** [comment]
Hi @eroom1966, thanks for the review.  I've updated the Makefile to clone the test-suite from https://github.com/riscv/riscv-compliance and will use that going forward.

@strichmo, I am going to request you to review this now just to ensure I didn't break anything wrt the new Makefile structures.

File: cv32/sim/uvmt_cv32/dsim.mk [hunk 1]
```diff
@@ -354,5 +379,6 @@ clean:
 	rm -rf $(DSIM_RESULTS)
 
 # All generated files plus the clone of the RTL
-clean_all: clean clean_core_tests clean_riscv-dv clean_test_programs clean-bsp
+clean_all: clean clean_core_tests clean_riscv-dv clean_test_programs clean_compliance
 	rm -rf $(CV32E40P_PKG)
```
Comment (strichmo) [inline cv32/sim/uvmt_cv32/dsim.mk:383]: Please restore the clean-bsp target as well as the clean_compliance

File: ci/run_compliance [hunk 2]
```diff
@@ -0,0 +1,122 @@
+#! /usr/bin/env python3
```
Comment (strichmo) [inline ci/run_compliance:1]: What's the primary reason for having a separate script?  Are there a lot of compliance tests?  I was hoping that we could fold these into our current flow without need of a separate script.

**Review by strichmo — CHANGES_REQUESTED** [review]

File: cv32/sim/uvmt_cv32/dsim.mk (hunk 1 above)
Comment (MikeOpenHWGroup) [reply cv32/sim/uvmt_cv32/dsim.mk:383]: Done!

File: ci/run_compliance (hunk 2 above)
Comment (MikeOpenHWGroup) [reply ci/run_compliance]: Awk!   This is why we have reviews of pull-requests.    That file was not supposed to get pushed-in.  I hacked-it up quickly to write out the compliance regression in `.metrics.json format`.  Nobody was ever supposed to see it...

**Comment by MikeOpenHWGroup** [comment]
Thanks for the review @strichmo.  I've pushed in fixes.

**Review by strichmo — APPROVED** [review]
Thanks!

**Pull request merged**